opengever/maintenance/scripts/set_iworks_mimetypes.py

```diff
@@ -0,0 +1,50 @@
+import mimetypes
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from os.path import splitext
+from plone import api
+import transaction
+
+
+def set_iworks_mimetype():
+    """Search all MS OneNote documents and fixes the contentType.
+    """
+
+    catalog = api.portal.get_tool('portal_catalog')
+    brains = catalog.unrestrictedSearchResults(
+        portal_type='opengever.document.document')
+    # A getIcon value of 'application.png' indicates that the mimetype
+    # coulnd't be resolved for this document - only consider these
+    brains = filter(lambda b: b.getIcon == 'application.png', brains)
+    for brain in brains:
+        obj = brain.getObject()
+        fname = obj.file.filename
+        ext = splitext(fname)[1]
+        if ext in [".numbers", ".pages", ".key"]:
+            obj.file.contentType = mimetypes.guess_type(fname)[0]
+            obj.reindexObject()
```

Comment:
please only reindex the affected indices. reindexing is expensive, especially `SearchableText` as this involves file content.

If there is no index and you're interested in metadata only just pick a cheap index like `id`.